package-res/resources/web/common-ui-require-js-cfg.js

```diff
@@ -63,6 +64,14 @@
   requirePaths["pentaho/_globalContextVars"] = basePath + "/pentaho/_globalContextVars";
   requirePaths["pentaho/contextVars"] = requirePaths["pentaho/_globalContextVars"];
 
+  // Singleton interface, virtual module ids
+  //requireMap["*"]["pentaho/spec/IContext"] = "pentaho/service!pentaho.spec.IContext?single";
+  requireMap["*"]["pentaho/config/IService"] = "pentaho/service!pentaho.config.IService?single";
+
+  // Default implementations
+  requireService["pentaho/config/impl/serviceInstance"] = "pentaho.config.IService";
+  requireService["pentaho/config/impl/AmdLoadedService"] = "Class<pentaho.config.IService>";
+
```

Comment:
We don't necessarily need to use the same nomenclature but on the server side that is the  guideline that is being used. We include both *api* and *impl* explicitly in the namespace (e.g. _org.pentaho.det.api.query.model.IDataSource_ / _org.pentaho.det.impl.query.model.DataSource_ ).